internal/armada/server.go

```diff
@@ -143,19 +163,24 @@ func Serve(config *configuration.ArmadaConfig, healthChecks *health.MultiChecker
 
 	grpc_prometheus.Register(grpcServer)
 
+	// TODO grpcCommon.Listen is supposed to call wg.Done() internally.
+	// Except, that it calls log.Fatalf first, so wg.Done() is never called.
```

Comment:
ah I just stepped through the code. Log.Fatal exits there and then.  Wow, that's not what I would have expected.

Reply:
`log.Fatal` calls `os.Exit(1)` internally, causing the program to exit immediately. For this reason, I think we should avoid using `log.Fatal`, with the possible exception of in the `main` function.
https://pkg.go.dev/log#Fatal

In this instance, closing the gRPC server with `Stop` or `GracefulStop` won't return an error, so the comment is not entirely accurate. However, if the server encounters an error, it will log at fatal, which causes the program to exit immediately without letting cleanup routines in other goroutines finish.

Reply:
yeah that's insane.  I'd be tempted to  never use log fatal in that case!

Reply:
Agreed :)